Tone/core/context/Context.ts

```diff
@@ -116,26 +109,30 @@ export class Context extends BaseContext {
 
 		if (options.context) {
 			this._context = options.context;
+			// custom context provided, latencyHint unknown (unless explicitly provided in options)
+			this._latencyHint = arguments[0]?.latencyHint || "";
 		} else {
 			this._context = createAudioContext({
 				latencyHint: options.latencyHint,
 			});
+			this._latencyHint = options.latencyHint;
 		}
 
 		this._ticker = new Ticker(
 			this.emit.bind(this, "tick"),
 			options.clockSource,
-			options.updateInterval
+			options.updateInterval,
+			this._context.sampleRate
 		);
 		this.on("tick", this._timeoutLoop.bind(this));
 
 		// fwd events from the context
 		this._context.onstatechange = () => {
 			this.emit("statechange", this.state);
 		};
-
-		this._setLatencyHint(options.latencyHint);
-		this.lookAhead = options.lookAhead;
+		
+		// if no custom updateInterval provided, updateInterval will be derived by lookAhead setter
+		this[arguments[0]?.hasOwnProperty("updateInterval") ? "_lookAhead" : "lookAhead"] = options.lookAhead;
```

Comment:
I'm not fully following this line. Is there a reason not just to set `this.lookAhead = options.lookAhead`

Reply:
Yes: The `lookAhead` setter I added also sets `updateInterval`. This line allows the dev to specify an `updateInterval` in the options if desired and not have it overridden by the `lookAhead` setter. In other words: If there is an `updateInterval` argument, set the private `_lookAhead` directly, otherwise use the `lookAhead` setter and allow that to also set `updateInterval`. This tries to cover all bases for example if the user wants a non-default `lookAhead` but didn't bother to specify any `updateInterval`.